aten/src/ATen/native/cpu/Reduce.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <ATen/native/cpu/Loops.h>
+#include <ATen/Parallel.h>
+
+#include <sstream>
+
+namespace at { namespace native { namespace {
+
+using namespace vec256;
+
+// reduction that is contiguous over the input in dim 0
+template <typename traits>
+static inline bool is_contiguous_reduction(const int64_t* strides) {
+  return strides[0] == 0 &&
+         strides[1] == sizeof(typename traits::arg2_t);
+}
+
+// reduction that is contiguous over the input in dim 1
+template <typename traits>
+static inline bool is_outer_reduction(const int64_t* strides) {
+  return strides[0] == 0 &&
+         strides[2] == sizeof(typename traits::result_type) &&
+         strides[3] == sizeof(typename traits::arg2_t);
+}
+
+template <typename func_t, typename vec_func_t>
+void binary_kernel_reduce_vec(TensorIterator& iter, func_t op, vec_func_t vop, int ident=0) {
```

Comment:
question: if we want to apply `TensorIterator` to `min`/`max` reduction, shouldn't we use `ident=-inf / +inf`?
In this case, would it be better to make `ident` of float type to support inf?

Reply:
Yeah, that makes sense